### PR TITLE
add tutorial notebooks conversion to onnx

### PR DESCRIPTION
**Pull request by geek-yang**
- Add tutorial notebooks about conversions from pytorch/tensorflow/keras/sklearn to onnx
- Add `requirements.txt` to list all the dependencies for notebooks about conversions

These notebooks are created based on the exploration (https://github.com/dianna-ai/dianna-exploration/tree/main/onnx_conversion_scripts).

### CODE REVIEW

**Comment by egpbos** [comment]
Is this ready for review?

**Comment by geek-yang** [comment]
Note that the `keras2onnx` notebook is slightly different from the one in the `dianna.exploration` repo. The old notebook uses `keras2onnx` package to convert `keras` model to `onnx` format. But the development of [`keras2onnx`](https://github.com/onnx/keras-onnx) has been stopped and the authors redirect all the users to `tf2onnx` package. Now we use `tf2onnx` package to do the conversion.

**Comment by geek-yang** [comment]
resnet files ([`resnet_savedmodel.onnx`](https://github.com/dianna-ai/dianna/blob/127-notebook-conversion-onnx/tutorials/models/resnet_savedmodel.onnx and [`resnet_savedmodel` folder](https://github.com/dianna-ai/dianna/tree/127-notebook-conversion-onnx/tutorials/models/resnet_savedmodel))) are saved to pass the test for notebooks. However, these two files are too large (over ~100 MB each). Shall we remove the files and skip the test for the related code blocks?

File: tutorials/conversion_onnx/keras2onnx.ipynb [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+      "2022-02-01 14:11:11,764 - INFO - \n",
+      "2022-02-01 14:11:11,764 - INFO - Successfully converted TensorFlow model ../models/resnet_savedmodel to ONNX\n",
+      "2022-02-01 14:11:11,765 - INFO - Model inputs: ['input_1']\n",
+      "2022-02-01 14:11:11,765 - INFO - Model outputs: ['predictions']\n",
+      "2022-02-01 14:11:11,765 - INFO - ONNX model is saved at ../models/resnet_savedmodel.onnx\n"
+     ]
+    }
+   ],
+   "source": [
+    "onnx_savedmodel = '../models/resnet_savedmodel.onnx'\n",
```
Comment (geek-yang) [inline tutorials/conversion_onnx/keras2onnx.ipynb:166]: This is the code which requires the model. Is it possible to skip the test only for this line in the notebook?

File: tutorials/conversion_onnx/keras2onnx.ipynb [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "INFO:tensorflow:Assets written to: ../models/resnet_savedmodel/assets\n"
+     ]
+    }
+   ],
+   "source": [
+    "savedmodel_dir = '../models/resnet_savedmodel'\n",
```
Comment (geek-yang) [inline tutorials/conversion_onnx/keras2onnx.ipynb:122]: Same here.

**Comment by loostrum** [comment]
> resnet files ([`resnet_savedmodel.onnx`](https://github.com/dianna-ai/dianna/blob/127-notebook-conversion-onnx/tutorials/models/resnet_savedmodel.onnx and [`resnet_savedmodel` folder](https://github.com/dianna-ai/dianna/tree/127-notebook-conversion-onnx/tutorials/models/resnet_savedmodel))) are saved to pass the test for notebooks. However, these two files are too large (over ~100 MB each). Shall we remove the files and skip the test for the related code blocks?

That explains why I suddenly had to download 180 MB when updating my copy of the repo :P 
I'm a bit surprised we have to store the models manually. Sure the converter might need to store it, but it should automatically download and use the keras builtin resnet. 

In any case I'd vote for removing the models (do check how to remove them completely from git history, otherwise they'll remain in the repo unless you do a shallow clone), and skipping (part of) the notebook if required

**Comment by geek-yang** [comment]
> > resnet files ([`resnet_savedmodel.onnx`](https://github.com/dianna-ai/dianna/blob/127-notebook-conversion-onnx/tutorials/models/resnet_savedmodel.onnx and [`resnet_savedmodel` folder](https://github.com/dianna-ai/dianna/tree/127-notebook-conversion-onnx/tutorials/models/resnet_savedmodel))) are saved to pass the test for notebooks. However, these two files are too large (over ~100 MB each). Shall we remove the files and skip the test for the related code blocks?
> 
> That explains why I suddenly had to download 180 MB when updating my copy of the repo :P I'm a bit surprised we have to store the models manually. Sure the converter might need to store it, but it should automatically download and use the keras builtin resnet.
> 
> In any case I'd vote for removing the models (do check how to remove them completely from git history, otherwise they'll remain in the repo unless you do a shallow clone), and skipping (part of) the notebook if required

@loostrum I couldn't agree more. We'd better move the large models as they make the repo too heavy...I need to figure out how to skip part of the test (if not possible, then the whole test for the keras conversion notebook). Any one has a quick solution to it?

**Comment by geek-yang** [comment]
> Is this ready for review?

Yes, it is now!

**Comment by geek-yang** [comment]
We can skip cells by throwing an exception (https://github.com/treebeardtech/nbmake). Need to test it.

**Review by egpbos — CHANGES_REQUESTED** [review]
Looks very good to me!

One final change I'd like to see is the `../models/` parts of file paths removed in `keras2onnx.ipynb` and `tensorflow2onnx.ipynb`. These are only sensible when running Jupyter from within the tutorials folder, which not everybody may do, so I would just remove the path and just stick to bare filenames, then people can pick their own working directory (= data dumping directory).

**Comment by geek-yang** [comment]
> Looks very good to me!
> 
> One final change I'd like to see is the `../models/` parts of file paths removed in `keras2onnx.ipynb` and `tensorflow2onnx.ipynb`. These are only sensible when running Jupyter from within the tutorials folder, which not everybody may do, so I would just remove the path and just stick to bare filenames, then people can pick their own working directory (= data dumping directory).
> 
> Feel free to merge after that, if nobody else objects ;)

Thanks for your comment @egpbos , and your help with the test :) I will remove the `../models/` parts. It is not necessary, I agree.

**Review by egpbos — APPROVED** [review]

**Pull request merged**